apps/accounts/validators.py

```diff
@@ -0,0 +1,237 @@
+                        raise ValidationError(
+                            ("You can not use a password that is already"
+                             " used in this application within password re-use interval [days hh:mm:ss]: {}.")
+                            .format(str(datetime.timedelta(seconds=self.password_reuse_interval))),
+                            code='password_used'
+                        )
+            except PastPassword.DoesNotExist:
+                pass
+
+            if self.password_min_age > 0 and passwds is not None and passwds.first() is not None:
+                if (datetime.datetime.now(datetime.timezone.utc)
+                        - passwds.first().date_created).total_seconds() <= self.password_min_age:
+                    # change password too soon
+                    raise ValidationError(
+                        "You can not change password that does not satisfy minimum password age [days hh:mm:ss]: {}."
+                        .format(str(datetime.timedelta(seconds=self.password_min_age))),
+                        code='password_used'
+                    )
+
+    def password_changed(self, password, user=None):
+
+        if not user or getattr(user, 'pk', None) is None or isinstance(getattr(user, 'pk', None), property):
+            warnings.warn('Change password on invalid user: {}'.format(user))
+            return
+
+        iter_val = PasswordHasher().iterations
+        userpassword_desc = UserPasswordDescriptor.objects.filter(
+            user=user,
+            iterations=iter_val
+        ).first()
+
+        if not userpassword_desc:
+            userpassword_desc = UserPasswordDescriptor()
+            userpassword_desc.user = user
+            userpassword_desc.save()
+
+        password_hash = userpassword_desc.create_hash(password)
+
+        # We are looking hash password in the database
+        tz_now = datetime.datetime.now(datetime.timezone.utc)
+        try:
+            # with the timestamp now() this look up will certainly not able to get an entry
+            # this is expected for new entry and re-use password (same user + password hash)
+            # note, re use of user + password hash will satisfy re use interval first.
+            PastPassword.objects.get(
+                userpassword_desc=userpassword_desc,
+                password=password_hash,
+                date_created=tz_now
+            )
+        except PastPassword.DoesNotExist:
+            past_password = PastPassword()
+            past_password.userpassword_desc = userpassword_desc
+            past_password.password = password_hash
+            past_password.save()
+
+    def get_help_text(self):
+        help_msg = ('Change password not allowed within mimimum password age [days hh:mm:ss]: {}, and'
+                    ' your new password can not be identical to any of the '
+                    'previously entered in the past [days hh:mm:ss] {}').format(
```

Comment:
```suggestion
        help_msg = ('For security, you can not change your password again for [days hh:mm:ss]: {}, and'
                    ' your new password can not be identical to any of the '
                    'previously entered in the past [days hh:mm:ss] {}').format(
```

Reply:
I don't know if that works better. Let me know what you all think.

Reply:
like the text, thanks French!